### PR TITLE
CLOUDSTACK-9928: Allow native CloudStack users to change password in UI

### PR DESCRIPTION
**Pull request by rhtyd**
This allows native CloudStack users to change password in UI when LDAP
is enabled. Overall changes:
- A new usersource returned in the listUsers response
- Removed ldap check in the UI, replaced with check based on user source
- DB changes to include user.source in user_view
- Changed UI error message for non-native users trying to change password

This also fixes `component/test_accounts.py`, adds a test case and includes that in Travis.

Pinging for review - @DaanHoogland @abhinandanprateek @resmo @wido @karuturi  and others

Screenshot with error for a Ldap user:
screenshot from 2017-05-25 16-41-03

Screenshot for normal user:
screenshot from 2017-05-25 16-41-40

### CODE REVIEW

**Comment by resmo** [comment]
code LGTM

@rhtyd I noticed the _error_ message in your screenshot is different (and much better) than 

> Error could not change your password because LDAP is enabled.

I would suggest to take your message as new and remove the existing (not replacing translations).

File: ui/l10n/en.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ var dictionary = {"ICMP.code":"ICMP Code",
 "changed.item.properties":"Changed item properties",
 "confirm.enable.s3":"Please fill in the following information to enable support for S3-backed Secondary Storage",
 "confirm.enable.swift":"Please fill in the following information to enable support for Swift",
-"error.could.not.change.your.password.because.ldap.is.enabled":"Error could not change your password because LDAP is enabled.",
+"error.could.not.change.your.password.because.non.native.user":"Error could not change your password because user is not a native CloudStack user.",
```
Comment (rhtyd) [inline ui/l10n/en.js]: @resmo this is changed here ^^ for english translation, for others I won't know exactly what string/text to use in respective languages.

**Comment by rhtyd** [comment]
@karuturi can you consider reviewing/merging this?

**Comment by rhtyd** [comment]
Thanks @borisstoyanov I'll merge it now.

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@karuturi can you help merge this?

**Comment by rhtyd** [comment]
@karuturi this is a UI/API fix, can you consider merging this?

**Comment by rhtyd** [comment]
@karuturi fixed conflict, can you consider merging this thanks.

**Comment by rhtyd** [comment]
@karuturi can you consider merging this?

**Comment by rhtyd** [comment]
Merging this since it has received enough reviews and test results.

**Pull request merged**